wagtail/wagtailsearch/index.py

```diff
@@ -77,6 +78,34 @@ def get_indexed_instance(self):
         """
         return self
 
+    @classmethod
+    def _has_field(cls, name):
+        try:
+            cls._meta.get_field(name)
+            return True
+        except models.fields.FieldDoesNotExist:
+            return hasattr(cls, name)
+
+    @classmethod
+    def check(cls, **kwargs):
+        errors = super(Indexed, cls).check(**kwargs)
+        errors.extend(cls._check_fields_exist(**kwargs))
+        return errors
+
+    @classmethod
+    def _check_fields_exist(cls, **kwargs):
+        errors = []
+        for field in cls.get_search_fields():
+            message = '{name} in search_fields but not on {model}'
```

Comment:
Maybe reword this message slightly? ``{model}.search_fields contains field '{name}' but it doesn't exist``